### PR TITLE
Fixed AppVeyor build configuration and the issue on SQL Server

### LINKED ISSUE #3451: Failed AppVeyor build allowed to pass
Right now master has a failing test on AppVeyor, yet the build is marked as successful:
https://ci.appveyor.com/project/doctrine/dbal/builds/22107072/job/ga7em29v6vktoxh2

### PR DESCRIPTION
**Pull request by morozov**
|      Q       |   A
|------------- | -----------
| Type         | bug
| BC Break     | no

Fixes #3451.

### CODE REVIEW

**Comment by morozov** [comment]
This is odd. The build is still passing, however, there's a lot of [repositories using the same approach](https://github.com/search?q=ErrorActionPreference+filename%3A.appveyor.yml&type=Code). It's also suggested in the [documentation](https://www.appveyor.com/docs/build-configuration/).

**Comment by morozov** [comment]
An example of a failing job with the same AppVeyor configuration prior to the fix of SQL Server platform: [1.0.1445](https://ci.appveyor.com/project/doctrine/dbal/builds/22232966/job/vo0r62aiel8uodhm#L26153).

File: lib/Doctrine/DBAL/Platforms/SQLServerPlatform.php [hunk 1]
```diff
@@ -1278,7 +1278,7 @@ protected function doModifyLimitQuery($query, $limit, $offset = null)
         // Even if the TOP n is very large, the use of a CTE will
         // allow the SQL Server query planner to optimize it so it doesn't
         // actually scan the entire range covered by the TOP clause.
-        if (! preg_match('/^(\s*SELECT\s+(?:DISTINCT\s+)?)(.*)$/im', $query, $matches)) {
+        if (! preg_match('/^(\s*SELECT\s+(?:DISTINCT\s+)?)(.*)$/is', $query, $matches)) {
```
Comment (Ocramius) [inline lib/Doctrine/DBAL/Platforms/SQLServerPlatform.php]: Woah, nice catch! Annoying that Powershell doesn't have a stricter default

**Review by Ocramius — APPROVED** [review]

**Comment by Ocramius** [comment]
@morozov does this need backporting?

**Comment by morozov** [comment]
I've backported the AppVeyor configuration changes, not sure yet about SQL Server yet. The test passed before the PHPStan-related changes even with the invalid regex flag.

**Pull request merged**